### DISCUSSION TITLE
How to use react hook form for search bar?

### DISCUSSION THREAD

**Discussion by aliakbarazizi** [react-hook-form/react-hook-form#9070] [Q&A] [answered]
I want to create search bar. and as the user type I want to search in api after few ms delay.

But I don't see how to use it

I found this solution on stackoverflow but it will call onSubmit without delay

https://stackoverflow.com/a/70119332/1827594

So is it ok to write something like this

```js
const { handleSubmit, watch } = useForm();
const onSubmit = (data) => console.log(data);

useEffect(() => {
   onst subscription = watch(() => {
    if (timer) clearTimeout(timer);

    timer = setTimeout(() => {
      handleSubmit(onSubmit)();
    }, 1000);
});
  return () => subscription.unsubscribe();
}, [handleSubmit, watch]);
```

**Comment by bluebill1049** [comment, marked as answer]
Do you even need to use react hook form for this use case (watch API)? you can just attach onChange on the input and debounce the request right?

**Reply by aliakbarazizi** [reply]
I see, but I still wondering, is my solution right? I just want to understand this library.

**Reply by bluebill1049** [reply]
could you just do this?

```tsx
<input onChange={(e) => {
  fetch('/test?query=' + e.target.value)
}} />
```

and yes, you can subscribe to input value as what you got above.

**Reply by aliakbarazizi** [reply]
> could you just do this?
> 
> ```tsx
> <input onChange={(e) => {
>   fetch('/test?query=' + e.target.value)
> }} />
> ```
> 
> and yes, you can subscribe to input value as what you got above.

Thank you
